rules/ensure_default_tags_rule_test.go

```diff
@@ -5,73 +5,155 @@ import (
 
 	hcl "github.com/hashicorp/hcl/v2"
 	"github.com/terraform-linters/tflint-plugin-sdk/helper"
+	awsRules "github.com/terraform-linters/tflint-ruleset-aws/rules"
 )
 
-func Test_EnsureDefaultTagsRule_Fails_WithMissingAttribute(t *testing.T) {
-	test := struct {
+func Test_EnsureDefaultTagsRule(t *testing.T) {
+	tests := []struct {
 		Name     string
 		Content  string
+		Config   string
 		Expected helper.Issues
 	}{
-		Name: "issue found",
-		Content: `
-		provider "aws" {
-			region = "eu-west-1"
-		}`,
-		Expected: helper.Issues{
-			{
-				Rule:    NewEnsureDefaultTagsRule(),
-				Message: "default_tags is missing",
-				Range: hcl.Range{
-					Filename: "resource.tf",
-					Start:    hcl.Pos{Line: 2, Column: 3},
-					End:      hcl.Pos{Line: 2, Column: 17},
+		{
+			Name: "Succeeds_ForProvider_WithTagsPresent",
+			Content: `
+			provider "aws" {
+				region = "eu-west-1"
+				default_tags {
+					tags = {
+						team = "platform-engineering",
+					}
+				}
+			}
+			
+			resource "aws_instance" "ec2_instance" {
+				region = "eu-west-1"
+			}`,
+			Config: `
+			rule "ensure_default_tags" {
+			  enabled   = true
+			  tags		= ["team"]
+			}`,
```

Comment:
Would it make sense to create a test case with multiple config rules, and ensure they both still work together?

Reply:
I don't think so, these rules are practically completely independent except for residing in the same namespace. They will be running alongside of hundreds of default TFLint rules